### PR TITLE
[10.x] Improved retrieval logic in the `Illuminate\Cache\DynamoDbStore` class

### PR DESCRIPTION
**Pull request by miladev95**
This pull request introduces a more streamlined approach to the get method in the Cache class. It refactors the code to use early returns for better readability and reduced nesting, resulting in a more concise and maintainable codebase. The refactored method includes clearer conditional checks, minimizing the number of if statements and providing direct returns based on the data's existence and expiration. This enhancement aims to improve the overall readability and maintainability of the Cache class' retrieval logic.

### CODE REVIEW

File: src/Illuminate/Cache/DynamoDbStore.php [hunk 1]
```diff
@@ -102,21 +102,13 @@ public function get($key)
...
+        $value = $response['Item'][$this->valueAttribute]['S'] ?? $response['Item'][$this->valueAttribute]['N'] ?? null;
 
-        if (isset($response['Item'][$this->valueAttribute])) {
-            return $this->unserialize(
-                $response['Item'][$this->valueAttribute]['S'] ??
-                $response['Item'][$this->valueAttribute]['N'] ??
-                null
-            );
-        }
+        return isset($value) ? $this->unserialize($value) : null;
```
Comment (crynobone) [inline src/Illuminate/Cache/DynamoDbStore.php:111]: Is checking `isset()` still correct? as it always fallback to `null`.

Comment (AndrewMast) [reply src/Illuminate/Cache/DynamoDbStore.php:111]: I believe that `is_null()` or just plain old `=== null` be best in this situation.

Comment (crynobone) [reply src/Illuminate/Cache/DynamoDbStore.php:111]: What's the different between `return $this->unserialize(null):` vs `return null;`?

Comment (miladev95) [reply src/Illuminate/Cache/DynamoDbStore.php:111]: @crynobone how it always fallback to `null`? 

`return isset($value) ? $this->unserialize($value) : null;`

this checks if `$value` is set then unserialize it

Comment (AndrewMast) [reply src/Illuminate/Cache/DynamoDbStore.php:111]: I tested it (copied it into tinker so maybe not really) and passing null into `unserialize` throws an error.

Comment (AndrewMast) [reply src/Illuminate/Cache/DynamoDbStore.php:111]: @miladev95 Well, you just declared `$value` on the line above. So while `isset($value)` resolves to false if the value is null, it also seems a bit confusing when you see it assigned a value in the line before.

**Pull request closed without merging**